pywheel/shim/_binwrapper.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+"""
+A generic wrapper to access nmodl binaries from a python installation
+Please create a softlink with the binary name to be called.
+"""
+import os
+from distutils import sysconfig
+import stat
+import sys
+from pkg_resources import working_set
+from distutils.ccompiler import new_compiler
+from distutils.sysconfig import customize_compiler, get_config_var
+
+
+def _set_default_compiler():
+    """Set (dont overwrite) CC/CXX so that apps dont use the build-time ones"""
+    ccompiler = new_compiler()
+    customize_compiler(ccompiler)
+    # xcrun wrapper must bring all args
+    if ccompiler.compiler[0] == 'xcrun':
+        ccompiler.compiler[0] = get_config_var("CC")
+        ccompiler.compiler_cxx[0] = get_config_var("CXX")
+    os.environ.setdefault("CC", ccompiler.compiler[0])
+    os.environ.setdefault("CXX", ccompiler.compiler_cxx[0])
+
+
+def _config_exe(exe_name):
+    """Sets the environment to run the real executable (returned)"""
+
+    package_name = 'nmodl'
+
+    assert package_name in working_set.by_key, "NMODL package not found! Verify PYTHONPATH"
+    NMODL_PREFIX = os.path.join(working_set.by_key[package_name].location, 'nmodl')
+    NMODL_PREFIX_DATA = os.path.join(NMODL_PREFIX, '.data')
+    os.environ["NMODL_WRAPLIB"] = os.path.join(NMODL_PREFIX_DATA, 'libpywrapper.so')
+
+    # find libpython*.so in the system
+    v = sysconfig.get_config_vars()
+    fpaths = [os.path.join(v[pv], v['LDLIBRARY']) for pv in ('LIBDIR', 'LIBPL')]
+    os.environ["NMODL_PYLIB"] = list(filter(os.path.exists, fpaths))[0]
+
```

Comment:
Omar - as you might remember, we were looking at an elaborated script for finding python library. Above works but not always the case?